src/app/modules/item/pages/chapter-group-progress/chapter-group-progress.component.html

```diff
@@ -1,10 +1,18 @@
-<alg-group-progress-grid
-  *ngIf="group$ | async as group; else userChapterView"
-  [itemData]="itemData"
-  [group]="group"
->
-</alg-group-progress-grid>
-
-<ng-template #userChapterView>
-  <p>Chapter view for user</p>
-</ng-template>
+<ng-container *ngIf="state$ | async as state">
+  <alg-group-progress-grid
+    *ngIf="state.isReady"
+    [itemData]="itemData"
+    [group]="state.data"
+  >
+  </alg-group-progress-grid>
+
+  <div class="spinner" *ngIf="state.isFetching">
+    <p-progressSpinner [style]="{ width: '3.3333rem', height: '3.3333rem' }"></p-progressSpinner>
+  </div>
+
```

Comment:
and what about the error case? Not directly related to the issue, I know, but related with the use of the state and important.

Reply:
Fixed